app/assets/src/components/views/SampleUploadFlow/ReviewStep.jsx

```diff
@@ -23,10 +33,19 @@ const processMetadataRows = metadataRows =>
 class ReviewStep extends React.Component {
   state = {
     consentChecked: false,
-    showUploadModal: false,
+    projectMetadataFields: null,
     showLessDescription: true,
+    showUploadModal: false,
   };
 
+  async componentDidMount() {
```

Comment:
Lets keep lifecycle methods lean and not blocking.
You should move the fetch call to server to its own function and have `componentDidMount` return immediately.
Even if it works, since this is not a function you yourself call, it makes no sense to change its behavior.
Also, because you are not calling it directly, it make no sense to guarantee that you return a Promise.

Reply:
extracted 👍